zephyr/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2022 Legrand North America, LLC.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+
+cmake_minimum_required(VERSION 3.10 FATAL_ERROR)
+
+
+# Detect the platform reliably
+if(ZEPHYR_BASE)
+else()
+  message(FATAL_ERROR "ZEPHYR_BASE needs to be defined for Zephyr builds")
+endif()
+
+# Verify the module name is set correctly
+set(expected_module_name "mcuboot")
+if(NOT "${module_name}" STREQUAL ${expected_module_name})
+  message(FATAL_ERROR "A module configuration error has occurred.
+This module is named '${module_name}' when '${expected_module_name}' \
+is expected. 
+Verify `module.yml` contains the line 'name: ${expected_module_name}'.")
```

Comment:
Can this be better formatted, looks jagged.

Reply:
Of course.  Please point me to an example of what you would like.

Reply:
https://github.com/mcu-tools/mcuboot/pull/1489/files#diff-00695564eb9dc6eac91ebf6608e7b236de613e6e0a329565c92df88056db7645R30, within this review.